c10/test/util/complex_test_common.h

```diff
@@ -0,0 +1,464 @@
+#include <type_traits>
+#include <tuple>
+#include <sstream>
+#include <c10/util/complex.h>
+#include <c10/macros/Macros.h>
+#include <gtest/gtest.h>
+
+#if (defined(__CUDACC__) || defined(__HIPCC__))
+#define MAYBE_GLOBAL __global__
+#else
+#define MAYBE_GLOBAL
+#endif
+
+#define PI 3.141592653589793238463
+
+namespace memory {
+
+MAYBE_GLOBAL void test_size() {
+  static_assert(sizeof(c10::complex<float>) == 8, "");
+  static_assert(sizeof(c10::complex<double>) == 16, "");
+}
+
+MAYBE_GLOBAL void test_align() {
+  static_assert(alignof(c10::complex<float>) == 8, "");
+  static_assert(alignof(c10::complex<double>) == 16, "");
+}
+
+MAYBE_GLOBAL void test_pod() {
+  static_assert(std::is_standard_layout<c10::complex<float>>::value, "");
```

Comment:
why is it true?

Reply:
I didn't see any violations in the definition: https://en.cppreference.com/w/cpp/named_req/StandardLayoutType

But I am not sure because I don't fully understand this:
> - None of the base class subobjects has the same type as the first non-static data member (see empty base optimization), and, recursively, the first non-static data member of that data member if it has non-union class type, or all non-static data members of that data member if it has union type, or an element of that data member if it has array type, etc.